src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -456,7 +456,40 @@ public static string GetPlatformApiLevel (string platformApiLevelName)
 
 		public static string GetLibraryImportDirectoryNameForAssembly (string assemblyIdentName)
 		{
-			return string.Concat (new MD2Managed ().ComputeHash (Encoding.UTF8.GetBytes (assemblyIdentName)).Take (5).Select (b => b.ToString ("X02")));
+			if (assemblyIdentityMap == null) {
+				throw new Exception ("Mapping was not loaded!");
+			}
+			if (assemblyIdentityMap.ContainsKey (assemblyIdentName)) {
+				return assemblyIdentityMap [assemblyIdentName];
+			}
+			string id = string.Format ("{0}", assemblyIdentityMap.Count + 1);
+			assemblyIdentityMap.Add (assemblyIdentName, id);
+			return id;
+		}
+
+		static Dictionary<string, string> assemblyIdentityMap;
+
+		public static void LoadAssemblyIdentityMap (string mapPath)
+		{
+			assemblyIdentityMap = new Dictionary<string, string> ();
+			if (!File.Exists (mapPath))
+				return;
+			foreach (var s in File.ReadLines (mapPath)) {
+				var items = s.Split ('=');
```

Comment:
if we really think about it we don't need a dictionary since we are just using the index of the item. So we just add the item names to a list in order and save that to disk. Then we can just using `IndexOf` to get the id. 

I don't think there is any need to have a key=value.. apart from ease of reading.